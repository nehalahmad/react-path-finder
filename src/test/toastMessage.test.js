import React from 'react';
import ToastMessage from './../components/toastMessage';
import {shallow} from 'enzyme';

describe("<ToastMessage />", () => {
  it("Render toast with a message", () => {
    const toastMessageObj = shallow(
      <ToastMessage errorMessage="Oops! something went wrong!" />
    );
    expect(toastMessageObj).toMatchSnapshot();
  });

  it("Render toast without a message", () => {
    const toastMessageObj = shallow(
      <ToastMessage />
    );
    expect(toastMessageObj).toMatchSnapshot();
  });

  it("it'll close the toast message", () => {
    const setErrorMessageHandler = jest.fn();
    const errorMessage = "Oops! something went wrong!";

    const toastMessageObj = shallow(
      <ToastMessage errorMessage={errorMessage} setErrorMessageHandler={setErrorMessageHandler} />
    );
    expect(toastMessageObj).toMatchSnapshot();

    let crossBtn = toastMessageObj.find('button.close');
    expect(crossBtn.length).toEqual(1);

    crossBtn.simulate("click");
    expect(setErrorMessageHandler.mock.calls.length).toEqual(1);
  });
});