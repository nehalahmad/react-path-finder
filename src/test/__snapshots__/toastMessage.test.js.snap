// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonContainer /> button container renders with 'Re-submit' text 1`] = `ShallowWrapper {}`;

exports[`<ButtonContainer /> button container renders with 'Submit' text 1`] = `ShallowWrapper {}`;

exports[`<ToastMessage /> Render toast with a message 1`] = `ShallowWrapper {}`;

exports[`<ToastMessage /> Render toast with a message 2`] = `ShallowWrapper {}`;

exports[`<ToastMessage /> Render toast without a message 1`] = `ShallowWrapper {}`;

exports[`<ToastMessage /> it'll close  1`] = `ShallowWrapper {}`;

exports[`<ToastMessage /> it'll close the toast message 1`] = `ShallowWrapper {}`;
